soc/espressif/Kconfig.sysbuild

```diff
@@ -1,10 +1,144 @@
+	  adding the swap status area size when calculating overflow.
+
+choice ESP_MCUBOOT_MODE
+	prompt "Mode of operation"
+	default ESP_MCUBOOT_MODE_SWAP_SCRATCH if BOOTLOADER_MCUBOOT_ESPRESSIF
+	help
+	  The operating mode of MCUboot (which will also be propagated to the application).
+
+config ESP_MCUBOOT_MODE_SINGLE_APP
+	bool "Single slot"
+	select ESP_MCUBOOT_IMGTOOL_OVERWRITE_ONLY
+	help
+	  MCUboot will only boot slot0_partition placed application and does not care about other
+	  slots. In this mode application is not able to DFU its own update to secondary slot and
+	  all updates need to be performed using MCUboot serial recovery.
+
+config ESP_MCUBOOT_MODE_SWAP_USING_OFFSET
+	bool "Swap using offset"
+	select EXPERIMENTAL
+	select ESP_MCUBOOT_BOOTLOADER_MODE_HAS_NO_DOWNGRADE
+	help
+	  MCUboot expects slot0_partition and slot1_partition to be present in DT and application
+	  will boot from slot0_partition. MCUBOOT_BOOTLOADER_NO_DOWNGRADE should also be selected
+	  in main application if MCUboot has been built with MCUBOOT_DOWNGRADE_PREVENTION.
+
+config ESP_MCUBOOT_MODE_SWAP_USING_MOVE
+	bool "Swap using move"
+	select ESP_MCUBOOT_BOOTLOADER_MODE_HAS_NO_DOWNGRADE
+	help
+	  MCUboot expects slot0_partition and slot1_partition to be present in DT and application
+	  will boot from slot0_partition. MCUBOOT_BOOTLOADER_NO_DOWNGRADE should also be selected
+	  in main application if MCUboot has been built with MCUBOOT_DOWNGRADE_PREVENTION.
+
+config ESP_MCUBOOT_MODE_SWAP_SCRATCH
+	bool "Swap using scratch"
+	select ESP_MCUBOOT_BOOTLOADER_MODE_HAS_NO_DOWNGRADE
+	help
+	  MCUboot expects slot0_partition, slot1_partition and scratch_partition to be present in
+	  DT, and application will boot from slot0_partition. In this mode scratch_partition is
+	  used as temporary storage when MCUboot swaps application from the secondary slot to the
+	  primary slot.
+	  MCUBOOT_BOOTLOADER_NO_DOWNGRADE should also be selected in main application if MCUboot
+	  has been built with MCUBOOT_DOWNGRADE_PREVENTION.
+
+config ESP_MCUBOOT_MODE_OVERWRITE_ONLY
+	bool "Overwrite"
+	select ESP_MCUBOOT_IMGTOOL_OVERWRITE_ONLY
+	select ESP_MCUBOOT_BOOTLOADER_MODE_HAS_NO_DOWNGRADE
+	help
+	  MCUboot will take contents of secondary slot of an image and will overwrite primary slot
+	  with it. In this mode it is not possible to revert back to previous version as it is not
+	  stored in the secondary slot.
+	  This mode supports MCUBOOT_BOOTLOADER_NO_DOWNGRADE which means that the overwrite will
+	  not happen unless the version of secondary slot is higher than the version in primary
+	  slot.
+
+endchoice # ESP_MCUBOOT_MODE
+
+config ESP_MCUBOOT_BOOTLOADER_MODE_HAS_NO_DOWNGRADE
+	bool
+	help
+	  Selected mode supports downgrade prevention, where you cannot switch to
+	  an application with lower version than the currently running application.
+
+if ESP_MCUBOOT_BOOTLOADER_MODE_HAS_NO_DOWNGRADE
+config ESP_MCUBOOT_BOOTLOADER_NO_DOWNGRADE
```

Comment:
newline gap before/after if and endit

Reply:
not fixed

Reply:
Sorry, I thought you meant removing the newlines. I've pulled this part from Zephyr and it doesn't have them: https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/Kconfig.mcuboot#L280-L288

What is the correct style for Kconfig, newline after every `if` and before `endif` or no gap?

Reply:
```
if ESP_MCUBOOT_BOOTLOADER_MODE_HAS_NO_DOWNGRADE

config ESP_MCUBOOT_BOOTLOADER_NO_DOWNGRADE
	bool

endif # ESP_MCUBOOT_BOOTLOADER_MODE_HAS_NO_DOWNGRADE
```